/**
 * Generated by Agitar build: JUnitFactory Version 2.1.0.000576 (Build date: Oct 19, 2007) [2.1.0.000576]
 * JDK Version: 1.5.0_11
 *
 * Generated on Nov 8, 2007 6:25:39 PM
 * Time to generate: 00:14.136 seconds
 *
 */

package org.mentawai.converter;

import com.agitar.lib.junit.AgitarTestCase;

public class IntegerConverterAgitarTest extends AgitarTestCase {
    
    public Class getTargetClass()  {
        return IntegerConverter.class;
    }
    
    public void testConstructor() throws Throwable {
        new IntegerConverter();
        assertTrue("Test call resulted in expected outcome", true);
    }
    
    public void testConvert() throws Throwable {
        Integer result = (Integer) new IntegerConverter().convert(new Integer(32));
        assertEquals("result", 32, result.intValue());
    }
    
    public void testConvert1() throws Throwable {
        Integer result = (Integer) new IntegerConverter().convert(new Integer(0));
        assertEquals("result", 0, result.intValue());
    }
    
    public void testConvertThrowsConversionException() throws Throwable {
        try {
            new IntegerConverter().convert("X0");
            fail("Expected ConversionException to be thrown");
        } catch (ConversionException ex) {
            assertEquals("ex.getMessage()", "java.lang.NumberFormatException: For input string: \"X0\"", ex.getMessage());
            assertThrownBy(NumberFormatException.class, ex);
        }
    }
    
    public void testConvertThrowsNullPointerException() throws Throwable {
        try {
            new IntegerConverter().convert(null);
            fail("Expected NullPointerException to be thrown");
        } catch (NullPointerException ex) {
            assertNull("ex.getMessage()", ex.getMessage());
            assertThrownBy(IntegerConverter.class, ex);
        }
    }
}

